controllers/cloud_config_sync_controller_test.go

```diff
@@ -0,0 +1,205 @@
+package controllers
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func getRequestForObject(object client.Object) reconcile.Request {
+	return reconcile.Request{
+		NamespacedName: client.ObjectKeyFromObject(object),
+	}
+}
+
+var _ = Describe("Cloud config sync controller", func() {
+	var r *CloudConfigReconciler
+	var rec *record.FakeRecorder
+
+	var infraCloudConfig *corev1.ConfigMap
+	var managedCloudConfig *corev1.ConfigMap
+	var syncedCloudConfigMap *corev1.ConfigMap
+
+	ctx := context.Background()
+
+	targetNamespaceName := testManagementNamespace
+
+	infraCloudConfName := "test-config"
+	infraCloudConfKey := "foo"
+
+	infrastructureResource := &configv1.Infrastructure{
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: infraCloudConfName,
+				Key:  infraCloudConfKey,
+			},
+			PlatformSpec: configv1.PlatformSpec{
+				Type: "VSphere",
+			},
+		},
+	}
+
+	syncedConfigMapKey := client.ObjectKey{Namespace: targetNamespaceName, Name: cloudConfigMapName}
+
+	BeforeEach(func() {
+		By("Creating needed ConfigMaps")
+		infraCloudConfig = &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+			Name:      infraCloudConfName,
+			Namespace: openshiftConfigNamespace,
+		}, Data: map[string]string{infraCloudConfKey: "bar"}}
+		managedCloudConfig = &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+			Name:      managedCloudConfigMapName,
+			Namespace: openshiftManagedConfigNamespace,
+		}, Data: map[string]string{"cloud.conf": "bar"}}
+		Expect(cl.Create(ctx, infraCloudConfig)).To(Succeed())
+		Expect(cl.Create(ctx, managedCloudConfig)).To(Succeed())
+
+		By("Create reconciler")
+		rec = record.NewFakeRecorder(32)
+		r = &CloudConfigReconciler{
+			Client:                 cl,
+			Scheme:                 scheme.Scheme,
+			Recorder:               rec,
+			InfrastructureResource: infrastructureResource,
+			TargetNamespace:        targetNamespaceName,
+		}
+
+		// First reconcile
+		_, err := r.Reconcile(ctx, getRequestForObject(infraCloudConfig))
+		Expect(err).Should(Succeed())
```

Comment:
Why are we manually calling reconcile instead of running it in the background like a manager in other tests?